stream-chat-android-client/src/main/java/io/getstream/chat/android/client/clientstate/SocketStateService.kt

```diff
@@ -54,6 +54,7 @@ internal class SocketStateService {
             state<SocketState.Disconnected> {
                 onEvent<ClientStateEvent.DisconnectedEvent> { _, _ -> stay() }
                 onEvent<ClientStateEvent.DisconnectRequestedEvent> { _, _ -> SocketState.Idle }
+                onEvent<ClientStateEvent.ConnectionRequested> { _, _ -> SocketState.Pending }
```

Comment:
The same as described above we shouldn't reinit connection manually when we were disconnected

Reply:
If the app is killed by the OS, the websocket will lose the connection and enter in this state. A user that sets the connection in an activity will see a crash if he/she doesn't manage it in its own side. Maybe this is a state that shouldn't happen... but it seams too much that the app crashes if the user asks for another connection when the app comes back in the foreground. 

Maybe we should deal log it or deal with it in another way? Let's not throw an exception here.